### ISSUE TITLE
Accept empty DATA_SOURCE_NAME

### ISSUE THREAD

**Issue by dominikh** [wrouesnel/postgres_exporter#28]
lib/pq supports various environment variables that make it possible to connect to a database with an empty dsn, by specifying the environment variables PGHOST, PGUSER, PGPASSWORD and so on. It would be nice if postgres_exporter, too, supported this.

Right now, one can work around this by using a value of `" "`, but that feels rather clumsy.

**Comment by jar349** [comment]
I used a value of `" "` and set PGHOST, PGUSER, and PGPASSWORD but was not able to connect.  Are you still able to use the work-around?

**Comment by jar349** [comment]
My motivation for having an empty DATA_SOURCE_NAME is that in my case I don't know the password because it's in a kubernetes Secret.  

See here: https://github.com/kubernetes/charts/blob/master/stable/postgresql/templates/deployment.yaml#L66-L85

Note that I can mount the password as an environment variable like so:
```
env:
        - name: DATA_SOURCE_NAME
          value: ""
        - name: PGHOST
          value: 127.0.0.1
        - name: PGUSER
          value: {{ default "postgres" .Values.postgresUser | quote }}
        - name: PGPASSWORD
          valueFrom:
            secretKeyRef:
              name: {{ template "postgresql.fullname" . }}
              key: postgres-password
```

But I don't have access to the password at configuration time.

**Comment by wrouesnel** [comment]
This was resolved by 528e347b48fe6e8a12027997a2bfabf4f815ea47. But a documentation update might be needed - the current variables available are
* username: `DATA_SOURCE_USER`
* password: `DATA_SOURCE_PASS`
* DSN: `DATA_SOURCE_NAME` or `DATA_SOURCE_URI`
